execinfrapb: panic in test builds when encountering malformed metadata

asubiotto opened the pull request:
Closes #60002 

We've seen cases where no metadata fields are set, causing a panic because we
assume that at least one field is set. This commit adds a panic in test builds
if this is the case and adds some protection against malformed metadata by
ignoring it if no fields are set.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/60734)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/60734#-:-MTq5yJhAyj4yb1ks69g:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

Commit:
execinfrapb: panic in test builds when encountering malformed metadata

We've seen cases where no metadata fields are set, causing a panic because we
assume that at least one field is set. This commit adds a panic in test builds
if this is the case and adds some protection against malformed metadata by
ignoring it if no fields are set.

Release note: None

asubiotto commented:
TFTR

bors r=yuzefovich
